test/built-ins/Set/GetSetRecord/size-is-not-undefined.js

```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2023 Anthony Frehner. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: pending
+description: GetSetRecord if the Set-like object has a size of 'undefined' an error is thrown
+info: |
+    2. Let rawSize be ? Get(obj, "size").
+    3. Let numSize be ? ToNumber(rawSize).
+    4. NOTE: If rawSize is undefined, then numSize will be NaN.
+    5. If numSize is NaN, throw a TypeError exception.
+features: [Set-methods]
+---*/
+
+assert.throws(
+  TypeError,
+  function () {
+    const s1 = new Set([1, 2]);
+    const s2 = {
+      size: undefined,
```

Comment:
Consider testing more inputs that yield NaN (but either in another file or rename this one)